src/commands/createFunction/genericSteps/StringPromptStep.ts

```diff
@@ -7,25 +7,33 @@ import { AzureWizardPromptStep } from "vscode-azureextensionui";
 import { ext } from "../../../extensionVariables";
 import { localize } from "../../../localize";
 import { IFunctionSetting } from "../../../templates/IFunctionSetting";
-import { IFunctionWizardContext } from "../IFunctionWizardContext";
+import { IBindingWizardContext } from "../../addBinding/IBindingWizardContext";
 
-export class StringPromptStep extends AzureWizardPromptStep<IFunctionWizardContext> {
+export class StringPromptStep extends AzureWizardPromptStep<IBindingWizardContext> {
     private readonly _setting: IFunctionSetting;
     constructor(setting: IFunctionSetting) {
         super();
         this._setting = setting;
     }
 
-    public async prompt(wizardContext: IFunctionWizardContext): Promise<void> {
+    public async prompt(wizardContext: IBindingWizardContext): Promise<void> {
         wizardContext[this._setting.name] = await ext.ui.showInputBox({
             placeHolder: this._setting.label,
             prompt: this._setting.description || localize('stringSettingPrompt', 'Provide a \'{0}\'', this._setting.label),
-            validateInput: (s: string): string | undefined => this._setting.validateSetting(s),
-            value: this._setting.defaultValue
+            validateInput: async (s): Promise<string | undefined> => await this.validateInput(wizardContext, s),
```

Comment:
nit: No typedef for s

Reply:
Yeah I turned that rule off a while ago because it was too annoying: https://github.com/Microsoft/vscode-azurefunctions/blob/master/tslint.json#L78

This specific case is a little weird, but it's mostly for cases [like this](https://github.com/Microsoft/vscode-azurefunctions/blob/1200241b475bad976204892b68c3949f33a7db8b/src/debug/getFuncTaskCommand.ts#L22):
```typescript
const x: string[] = ['hello', 'world'];
x.find((s: string) => s === 'world'); // <-- gross. and obvious that s will be a string
x.find(s => s === 'world'); // <-- 😊
```

Let me know if you strongly feel we should turn that back on.